pandas/_libs/tslibs/parsing.pyx

```diff
@@ -386,7 +386,7 @@ cdef parse_datetime_string_with_reso(
         &out_tzoffset, False
     )
     if not string_to_dts_failed:
-        if dts.ps != 0 or out_local:
+        if out_bestunit == NPY_DATETIMEUNIT.NPY_FR_ns or out_local:
```

Comment:
makes sense.  to be totally correct we might need to check for ps, fs, as?

Reply:
Good point. What should be the error message we raise, or should we silently ignore?

On 1.5.2, doing ``pd.Period("1970/01/01 00:00:00.000000000111")`` gives me
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pandas/_libs/tslibs/period.pyx", line 2579, in pandas._libs.tslibs.period.Period.__new__
  File "pandas/_libs/tslibs/parsing.pyx", line 369, in pandas._libs.tslibs.parsing.parse_time_string
  File "pandas/_libs/tslibs/parsing.pyx", line 431, in pandas._libs.tslibs.parsing.parse_datetime_string_with_reso
KeyError: 11
```

Timestamp seems to ignore the extra precision, so maybe we should match that here?

Reply:
definitely better than the status quo.  it might be reasonable to have Timestamp warn/raise instead of silently truncating, but thats out of scope.